app/routes.py

```diff
@@ -1,2 +1,412 @@
+                up_one_count += 1
+
+    response = {
+            'customer_id': customer_id,
+            'video_id': video_id,
+            'videos_checked_out_count': down_one_count,
+            'available_inventory': up_one_count 
+            }
+    return jsonify(response), 200
+
+# the rental should be linked to a customer and a video -- 
+# both the customer and the video 
+    # if rental is None:
+    #     return jsonify(None), 400
+
+######################## GET RENTAL /customers/<id>/rentals CRUD - READ ###################
+@customers_bp.route("/<id>/rentals", methods=["GET"])
+def list_all_videos_customer_has_out(id):
+    customer_id = id
+    customer = Customer.query.get(customer_id)
+    customers_current_videos = customer.rentals # accesses the rentals column in the data customer table # get this row with the rental_id
+    video_id_list = []
+    for rental in customers_current_videos:
+        video_id_list.append(rental.video_id)
+    print(video_id_list)
+
+    response = []
+    for video_id in video_id_list:
+        rental_details = Video.query.get(video_id)
+        release_date = rental_details.release_date
+        title = rental_details.title
+        due_date = rental.due_date
+
+        video_info = {"release_date": release_date, "title": title, "due_date": due_date}
+        
+        response.append(video_info)
+    return jsonify(response), 200
+
+# db.session.query(Video, Customer, Rental).join(Video, Video.id==Rental.video_id).join(Customer, Customer.id==Rental.customer_id).filter(Video.id == who_has_the_vid).all()
+    # ^ result will be an array of tuples. Each tuple will hold 
+    # a Video instance, Customer instance, and a Rental instance 
+        # what_vids_with_customer = ???
+        # videos_with_customer 
+
+######################## GET RENTAL /videos/<id>/rentals CRUD - READ ###################
+@videos_bp.route("/videos/<id>/rentals", methods=["GET"])
+def list_all_customers_who_currently_have_video(id):
+    video_id = id
+    # if not video_id.isdigit():
+    #     error_response2 = "is not a number"
+    #     return jsonify(errors={"video_id": error_response2}), 400
+
+    video = Video.query.get(video_id)
+    video_as_rental = video.rentals # accesses the rentals column in the data customer table # get this row with the rental_id
+    
+    customers_with_video_list = []
+    for rental in video_as_rental:
+        customers_with_video_list.append(rental.customer_id)
+
+    response = []
+    for customer_id in customers_with_video_list:
+        rental_customer_details = Customer.query.get(customer_id)
+        name = rental_customer_details.name
+        phone = rental_customer_details.phone
+        postal_code = rental_customer_details.postal_code
+        for rental in customer_id.rentals:
+            if rental.video_id == video_id:
+                due_date = rental.due_date # find the rental instance that matches the customer_id and the video_id and get that due date
```

Comment:
Combining these two for loops makes the inner for loop on lines 383-385 unnecessary.  The rental from 374 already has the due date.  Also, there's a small bug on 383, the customer_id is just the id, the rental_customer_details is the class instance with the rentals attribute.